source/slang/slang-check-expr.cpp

```diff
@@ -1355,6 +1355,162 @@ namespace Slang
         }
     }
 
+    RefPtr<Expr> SemanticsVisitor::CheckMatrixSwizzleExpr(
+        MemberExpr* memberRefExpr,
+        RefPtr<Type>      baseElementType,
+        IntegerLiteralValue baseElementRowCount,
+        IntegerLiteralValue baseElementColCount)
+    {
```

Comment:
On reading through this code it seems like this code could be improved from a readability point of view by using a struct to represent the coord. Doing so would make clearer the intent with 4 coordinates in an array. So something like

```
struct MatrixCoord
{
   bool operator==(...);
   bool operator!=(...) const { return !(*this == rhs); }
   // Rows and columns are zero indexed 
   int row;   
   int col;
};
```
You use the ability to access the row or col via an index. You could do with an accessor that returns a reference on the struct.